site/app/libraries/database/DatabaseQueries.php

```diff
@@ -149,8 +149,13 @@ public function buildLoadThreadQuery($categories_ids, $thread_status, $show_dele
             }
             $query_raw_select[]     = "t.*";
             $query_raw_select[]     = "({$query_favorite}) as favorite";
-            $query_raw_select[]     = "(case when exists(select 1 from posts p where p.author_user_id = sf.user_id and p.thread_id = t.id) then true else false end) as current_user_posted";
+            $query_raw_select[]     = "CASE
+                                        WHEN EXISTS(SELECT * FROM (posts p LEFT JOIN forum_posts_history fp ON p.id = fp.post_id AND p.author_user_id != fp.edit_author) AS pfp WHERE (pfp.author_user_id = ? OR pfp.edit_author = ?) AND pfp.thread_id = t.id) THEN true 
+                                        ELSE false 
+                                        END as current_user_posted";
 
+            $query_parameters[]     = $current_user;
+            $query_parameters[]     = $current_user;
```

Comment:
Is this a duplicate / unnecessary line?

Reply:
No, the line above adds `(pfp.author_user_id = ? OR pfp.edit_author = ?)` which we check for the `$current_user` in both of them.